### PR TITLE
Fuzzy search

### PR DESCRIPTION
**Pull request by sarthak-sehgal**
Implemented a "fuzzy search" through Vue for the components. A fuzzy search basically searches by breaking the query and not for the exact same query and assigns scores to the results, though, of course it gives maximum score to the result which matches exactly the query. I think that fuzzy search is important for a platform like BioJS since there are a lot of jargons and technical terms used so it will be really helpful for a user with not the best knowledge of bioinformatics.
Screenshots:
biojs week 5
biojs week 5 1

### CODE REVIEW

**Comment by rowlandm** [comment]
Can we get a URL when we create these PRs please? Especially URLS where we
can start to test the PR?

**Comment by Megh-Thakkar** [comment]
Hi @rowlandm, you can try the search bar [here](http://139.59.93.32/biojs-frontend/dist/#/components)

**Comment by rowlandm** [comment]
Cool. Thanks Sarthak!

**Comment by sarthak-sehgal** [comment]
@rowlandm is this good for now?

File: src/components/SearchComponent.vue [hunk 1]
```diff
@@ -1,12 +1,77 @@
...
+	computed: {
+		queryResults () {
+			const preparedQuery = fz.prepareQuery(this.query);
+			const scores = {};
+			return this.components.map((component, index) => {
+				const componentScores = [
+					component.id,
+					component.name,
+					component.tags.join(', ')
+				].map(toScore => fz.score(toScore, this.query, { preparedQuery }));
```
Comment (rowlandm) [inline src/components/SearchComponent.vue]: Would be good to write a comment about this one big line.

I would prefer this return line that is split across multiple lines be separate lines themselves. This makes it easier to debug.

File: src/components/SearchComponent.vue [hunk 2]
```diff
@@ -1,12 +1,77 @@
...
+
+				scores[component.id] = Math.max(...componentScores);
+
+				return component;
+			})
+				.filter(component => scores[component.id] > 1)
```
Comment (rowlandm) [inline src/components/SearchComponent.vue]: would be great to write a comment here.

**Comment by sarthak-sehgal** [comment]
@rowlandm, sure, I will add comments and also a link to the documentation of the 3rd party package.

**Comment by sarthak-sehgal** [comment]
@rowlandm please have a look at the latest commit. Is it good now?

**Pull request merged**